### PR TITLE
default-rule validators changes

### PR DESCRIPTION
**Pull request by sumanvpacewisdom**

### CODE REVIEW

File: src/validators/v1/default-rule.js [hunk 1]
```diff
@@ -54,6 +51,7 @@ module.exports = {
 				.withMessage('Target field is required and must be a string')
 
 			req.checkBody('is_target_from_sessions_mentor')
+				.optional()
```
Comment (nevil-mathew) [inline src/validators/v1/default-rule.js:54]: Remove checkBody for read, only id is needed

Comment (nevil-mathew) [reply src/validators/v1/default-rule.js:54]: and id is optional

File: src/validators/v1/default-rule.js [hunk 2]
```diff
@@ -21,7 +22,11 @@ module.exports = {
 
 			req.checkBody('field_configs').optional().isJSON().withMessage('Field configs must be a valid JSON')
 
-			req.checkBody('matching_operator').isString().notEmpty().withMessage('Matching operator must be a string')
+			req.checkBody('matching_operator')
+				.isString()
+				.notEmpty()
+				.isIn(['=', '!=', '>', '<', '>=', '<='])
```
Comment (nevil-mathew) [inline src/validators/v1/default-rule.js:28]: Update this to words, eg: "equals"

File: src/validators/v1/default-rule.js
Comment (nevil-mathew) [inline src/validators/v1/default-rule.js:1]: req.checkBody('role_config').isObject()

is .isObject() valid?

Comment (nevil-mathew) [reply src/validators/v1/default-rule.js:1]: and add isIn to type as well, with allowed values in session, and mentor

**Review by nevil-mathew — CHANGES_REQUESTED** [review]

**Comment by sumanvpacewisdom** [comment]
@nevil-mathew Updated the commented changes , please review and let me know

File: src/validators/v1/default-rule.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+					'equals',
+					'not equals',
+					'greater than',
+					'less than',
+					'greater than or equals',
+					'less than or equals',
+				])
+				.withMessage(
+					'Matching operator must be one of "equals", "not equals", "greater than", "less than", "greater than or equals", "less than or equals"'
+				)
```
Comment (nevil-mathew) [inline src/validators/v1/default-rule.js:42]: It should match from these:
	['equals', '='],
	['notEquals', '!='],
	['contains', '@>'],
	['containedBy', '<@'],
	['overlap', '&&'],
	['lessThan', '<'],
	['greaterThan', '>'],
	['lessThanOrEqual', '<='],
	['greaterThanOrEqual', '>='],
so please remove the spaces in between.

**Review by nevil-mathew — APPROVED** [review]

**Pull request merged**